src/bp/core/services/migration/migration_11_9_2.ts

```diff
@@ -0,0 +1,71 @@
+import * as sdk from 'botpress/sdk'
+import { BotConfig } from 'core/config/bot.config'
+
+import { ConfigProvider } from '../../config/config-loader'
+import { BotService } from '../bot-service'
+import { WorkspaceService } from '../workspace-service'
+
+import { MigrationStep } from '.'
+
+export default class Migration implements MigrationStep {
+  constructor(
+    private logger,
+    private configProvider: ConfigProvider,
+    private botService: BotService,
+    private workspaceService: WorkspaceService
+  ) {}
+
+  public async execute(dryRun?: boolean): Promise<sdk.MigrationStatus> {
+    const bots = await this.botService.getBots()
+    const pipeline = await this.workspaceService.getPipeline()
+
+    const hasChanges = await this._ensureBotConfigCorrect(bots, pipeline[0], dryRun)
+
+    return {
+      hasConfigChanges: hasChanges,
+      hasFileChanges: false
```

Comment:
why not return a list of descriptive changes instead? if array is empty, it means there are no fileChanges.  This would be more explicit both to the end-user and also for the developer.

example:

automaticConfigChanges: []
manualConfigChanges: []
automaticFileChanges: [ "A new property called `authentication` will be added to botpress.config.json" ]
manualFileChanges: [ "You need to open the file workspaces.json and assign a superAdmin manually" ]

Manual file changes crash the process (w/ proper exit code) until resolved

Reply:
I wasn't sure on how much overhead we wanted to add to future migrations. I agree that it's much better to add a list of changes, now it's a bit cryptic

Reply:
There's no such thing has having "too much overhead" while creating/running migrations. I prefer having the migration process to be as clear as possible.